sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1548,6 +1678,166 @@ different from the one with which the dynamic_parameter was created.
+for use with `event::get_info<>()` which can be used to query the type of a SYCL
+event.
+
+Table {counter: tableNumber}. Additions to the `sycl::event` class information descriptors.
+[%header,cols="2a,a,a"]
+|===
+|Event Descriptor
+|Return Type
+|Description
+
+|info::event::ext_oneapi_event_type
+|info::ext_oneapi_event_t
+|Returns the type of this SYCL event
+|===
+
+Events of type `ext_oneapi_event_t::normal` are normal SYCL events as described
+in the SYCL specification. `ext_oneapi_event_t::graph_limited` and
+`ext_oneapi_event_t::graph_external` events are new as part of this extension,
+and are differentiated from normal SYCL events in how they can be used.
+
+==== Graph-Limited Events
+
+Events associated with a graph node (those returned from recorded queue
+submissions) cannot be used as normal SYCL events outside of their associated
+graph, these are known as "graph-limited" events. They can only be used as
+parameters to `handler::depends_on()`, or as dependent events for queue
+shortcuts like `queue::parallel_for()` for other submissions recorded to
+the same modifiable `command_graph`.
+
+The following limitations apply to graph-limited events:
+
+- Calling `event::get_info<info::event::command_execution_status>()` or
+`event::get_profiling_info()` will throw synchronously with error code
+`invalid`.
+
+- A host-side wait on the event will throw synchronously with error
+code `invalid`.
+
+- Using the event outside of the recording scope will throw synchronously with
+error code `invalid`.
+
+==== External Events
+
+To overcome the limitations of "graph-limited" events, users can add external
+event nodes to a graph. These represent a point of execution within the graph
+and provide a SYCL event object which can be used to both manage synchronization
+between different graphs as well as between the graph and regular operations
+submitted to a SYCL queue. External graph events are associated with a single
+external event node inside a graph. External events are functionally normal SYCL
+events with some restrictions (see below).
+
+External events can be obtained by either:
+
+* Calling `command_graph::add_external_event()` on a modifiable `command_graph`
+to add an external event node, then obtain the SYCL event object by calling
+`node::get_external_event()`. 
+
+* Calling `queue::ext_oneapi_external_event()` on a queue in the recording
+state, which will return an external SYCL event.
+
+External events may be used as parameters to `handler::depends_on()` or as
+dependent events for queue submissions outside of the graph they
+are associated with. For example, this allows enqueing work outside of the graph
+part-way through the graph execution, rather than having to wait for the entire
+graph to finish.
+
+External events are considered complete when all their dependent graph nodes
+have finished execution, and their execution status is automatically reset when
```

Comment:
Since there is some other discussion going on around update which may affect these I'll leave this unresolved for now until we have more clarity on this aspect.

Reply:
I also thought this language about being automatically reset was too vague.  If you read this literally, then I think your example using the explicit API is not well defined:

```
// Add an external event node to Graph A
auto ExtEventNode = GraphA.add_external_event();
auto ExternalEventA = ExtEventNode.get_external_event();

// Add a wait external event node in GraphB
auto ExtWaitNode = GraphB.add_wait_external_event({ExternalEventA});

queueA.submit([&](handler& cgh) {
  cgh.ext_oneapi_graph(execGraphA);
});

// Is "ExternalEventA" really reset at this point?
// If so, then is the use of "ExternalEventA" in GraphB below ever signalled?

queueB.submit([&](handler& cgh) {
  cgh.ext_oneapi_graph(execGraphB);
});
```